File: samples/FortyTwoTest.java
package samples;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.junit.Assert.assertThat;

import org.junit.Test;

public class FortyTwoTest {
  private final FortyTwo fortyTwo = new FortyTwo();

  @Test
  public void testFortyTwo() {
    assertThat(fortyTwo.fortyTwo(), equalTo(42));
  }
}

File: samples/FortyTwoTest.hs
import Test.Framework
import Test.HUnit
import Test.Framework.Providers.HUnit
import Test.Framework.Providers.QuickCheck2

import Samples.FortyTwo

main = defaultMain tests

tests = [
        testGroup "Sorting Group 1" [
		testCase "fortytwo" test_fortytwo
            ]
    ]

test_fortytwo = fortytwo @=? 42
